src/metatensor/models/utils/data/dataset.py

```diff
@@ -1,13 +1,50 @@
 from dataclasses import dataclass
-from typing import Dict, List, NamedTuple, Optional, Tuple, Union
+from typing import Any, Dict, List, Optional, Tuple, Union
 
+import metatensor.learn
 import torch
-from metatensor.learn.data import Dataset, group_and_join
 from metatensor.torch import TensorMap
 from torch import Generator, default_generator
 from torch.utils.data import Subset, random_split
 
 
+class Dataset:
+    """A version of the `metatensor.learn.Dataset` class that allows for
+    the use of `mtm::` prefixes in the keys of the dictionary.
+
+    It is important to note that, instead of named tuples, this class
+    accepts and returns dictionaries.
+
+    :param dict: A dictionary with the data to be stored in the dataset.
+    """
```

Comment:
I assume that this is a temporary solution, right?

Reply:
It is a temporary solution, but we might have to use it for a few months because the fix to the metatensor.learn.Dataset class seems quite complicated to me

Reply:
Okay then maybe add a comment/internal note and a link to the issue in metatensor (if there is one).